Apps/WebClient/src/ClientApp/src/components/calendar/body.vue

```diff
@@ -175,62 +175,60 @@ export default class CalendarBodyComponent extends Vue {
     @Prop() private isVisible!: boolean;
 
     private eventLimit: number = 4;
-    private firstEventDateOfCurrentMonth: Date | null = null;
 
     private isHovering: boolean = false;
     private hoveringEvent: CalendarEntry | null = null;
     private eventBus = EventBus;
-    private get currentDates() {
-        return this.getCalendar();
-    }
 
-    private getCalendar(): CalendarWeek[] {
-        let now = new Date();
-        let current = new Date(this.currentDate);
+    private currentDates: CalendarWeek[] = [];
+
+    @Watch("currentDate")
+    private onCurrentDateUpdate() {
+        this.currentDates = this.getMonthCalendar(this.currentDate);
+
+        if (this.isVisible) {
+            let dateGroup: DateGroup = this.dateGroups.find((d) =>
+                moment(this.currentDate).isSame(d.date, "month")
+            ) as DateGroup;
+            this.eventBus.$emit(
+                EventMessageName.CalendarMonthChanged,
+                dateGroup.entries[0].date
+            );
+        }
+    }
 
-        let startDate = DateUtil.getMonthFirstDate(current);
+    private getMonthCalendar(monthDate: Date): CalendarWeek[] {
```

Comment:
btw, I'm not sure what is the issue or how hard is it to get the this.slotEvents out and pass it as the 2nd param here though. Regardless if it is class/object attribute or not, it can be passed from the caller here right? Code will be cleaner and easier if we decide to perform unit-test or centralize this method later one.

Reply:
it can be done, but it is unnecessary since the variable is passed as a property to the class. If unit testing, the variable would have to be set as well